client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FC } from 'react';
+import { StandardTextFieldProps } from '@mui/material';
+import { BasicTextInput } from './BasicTextInput';
+import { NumUtils } from '@common/utils';
+export interface NumericTextInputProps
+  extends Omit<StandardTextFieldProps, 'onChange'> {
+  onChange?: (value: number | undefined) => void;
+  width?: number | string;
+  defaultValue?: number;
+  allowNegative?: boolean;
+  min?: number;
+  max?: number;
+  precision?: number;
```

Comment:
:nit while I like how concise `precision` is, I think it requires explanation, where `decimalPlaces` or `decimalLimit` do not

Reply:
Agree, have renamed to `decimalLimit`